Tests/src/PublicAPITests/MultiPlayerTests.cpp

```diff
@@ -3040,4 +3040,224 @@ CSP_PUBLIC_TEST(CSPEngine, MultiplayerTests, CreateObjectParentTest)
+			 ChildEntityName2](SpaceEntity* Entity, SpaceEntityUpdateFlags Flags, csp::common::Array<ComponentUpdateInfo>& UpdateInfo)
+			{
+				if (Entity->GetName() == ChildEntityName2 && Flags & SpaceEntityUpdateFlags::UPDATE_FLAGS_PARENT)
+				{
+					ChildEntityUpdated = true;
+				}
+			});
+
+		CreatedChildEntity2->SetParentId(CreatedParentEntity->GetId());
+
+		CreatedChildEntity2->QueueUpdate();
+
+		WaitForCallbackWithUpdate(ChildEntityUpdated, EntitySystem);
+		EXPECT_TRUE(ChildEntityUpdated);
+	}
+
+	// Delete the parent
+	{
+		bool DestroyCalled = false;
+
+		auto DestroyCb = [&DestroyCalled](bool Success)
+		{
+			DestroyCalled = true;
+			EXPECT_TRUE(Success);
+		};
+
+		bool ChildEntityUpdated = false;
+
+		CreatedChildEntity1->SetUpdateCallback(
+			[&ChildEntityUpdated,
+			 ChildEntityName1](SpaceEntity* Entity, SpaceEntityUpdateFlags Flags, csp::common::Array<ComponentUpdateInfo>& UpdateInfo)
+			{
+				if (Entity->GetName() == ChildEntityName1 && Flags & SpaceEntityUpdateFlags::UPDATE_FLAGS_PARENT)
+				{
+					ChildEntityUpdated = true;
+				}
+			});
+
+		bool ChildEntityUpdated2 = false;
+
+		CreatedChildEntity2->SetUpdateCallback(
+			[&ChildEntityUpdated2,
+			 ChildEntityName2](SpaceEntity* Entity, SpaceEntityUpdateFlags Flags, csp::common::Array<ComponentUpdateInfo>& UpdateInfo)
+			{
+				if (Entity->GetName() == ChildEntityName2 && Flags & SpaceEntityUpdateFlags::UPDATE_FLAGS_PARENT)
+				{
+					ChildEntityUpdated2 = true;
+				}
+			});
+
+		EntitySystem->DestroyEntity(CreatedParentEntity, DestroyCb);
+
+		WaitForCallbackWithUpdate(DestroyCalled, EntitySystem);
+		EXPECT_TRUE(DestroyCalled);
+
+		WaitForCallbackWithUpdate(ChildEntityUpdated, EntitySystem);
+		EXPECT_TRUE(ChildEntityUpdated);
+
+		WaitForCallbackWithUpdate(ChildEntityUpdated2, EntitySystem);
+		EXPECT_TRUE(ChildEntityUpdated2);
+
+		// Check children are unparented correctly
+		EXPECT_EQ(CreatedChildEntity1->GetParentEntity(), nullptr);
+		EXPECT_EQ(CreatedChildEntity2->GetParentEntity(), nullptr);
+
+		EXPECT_EQ(EntitySystem->GetNumEntities(), 2);
+		EXPECT_EQ(EntitySystem->GetRootHierarchyEntities()->Size(), 2);
+	}
+
+	// Re-enter space to ensure updates were made to the server
+	{
+		// Exit Space
+		auto [ExitSpaceResult] = AWAIT_PRE(SpaceSystem, ExitSpace, RequestPredicate);
+
+		// Log out
+		LogOut(UserSystem);
+
+		// Log in again
+		LogIn(UserSystem, UserId);
+
+		// Enter space
+		bool EntitiesCreated = false;
+
+		auto EntitiesReadyCallback = [&EntitiesCreated](bool Success)
+		{
+			EntitiesCreated = true;
+			EXPECT_TRUE(Success);
+		};
+
+		EntitySystem->SetInitialEntitiesRetrievedCallback(EntitiesReadyCallback);
+
+		auto [EnterResult2] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+		EXPECT_EQ(EnterResult2.GetResultCode(), csp::systems::EResultCode::Success);
+
+		WaitForCallbackWithUpdate(EntitiesCreated, EntitySystem);
+		EXPECT_TRUE(EntitiesCreated);
+	}
+
+	// Ensure children have been parented to the root
```

Comment:
[Question] Is "parented to the root" the correct terminology, considering that `getParentEntity` is null now? Is this actually a deparent? Maybe :
```suggestion
	// Ensure children have been unparented and are now root entities
```

Reply:
This is better!